osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -251,8 +251,9 @@ private void removeSourceClockAdjustments()
 
         private class HardwareCorrectionOffsetClock : FramedOffsetClock
         {
-            // we always want to apply the same real-time offset, so it should be adjusted by the playback rate to achieve this.
-            public override double CurrentTime => SourceTime + Offset * Rate;
+            // we always want to apply the same real-time offset, so it should be adjusted by the difference in playback rate (from realtime) to achieve this.
+            // base implementation already adds offset at 1.0 rate, so we only add the difference from that here.
+            public override double CurrentTime => base.CurrentTime + Offset * (1 - Rate);
```

Comment:
Am I wrong in thinking this formula seems different to what it was before?

```
was: SourceTime + Offset * Rate

is:  FramedOffsetClock.CurrentTime + Offset * (1 - Rate) =
     FramedClock.CurrentTime + Offset + Offset * (1 - Rate) =
     FramedClock.CurrentTime + Offset * (2 - Rate) =
     SourceTime + Offset * (2 - Rate)
```

Think you might have meant `(Rate - 1)`?

Reply:
That was the intention, oops.